test/Conversion/TorchToTosa/constant_pad_nd.mlir

```diff
@@ -0,0 +1,17 @@
+// RUN: torch-mlir-opt <%s -convert-torch-to-tosa -split-input-file -verify-diagnostics | FileCheck %s
+
+// CHECK-LABEL:   func.func @torch.aten.constant_pad_nd(
+// CHECK-SAME:                                %[[ARG:.*]]: !torch.vtensor<[1,512,13,13],f32>) -> !torch.vtensor<[1,512,14,14],f32> {
+// CHECK:           %[[VAL_2:.*]] = "tosa.const"()
+// CHECK:           %[[VAL_3:.*]] = "tosa.const"() {value = dense<0xFF800000> : tensor<f32>} : () -> tensor<f32>
+// CHECK:           %[[VAL_4:.*]] = "tosa.pad"
```

Comment:
As this is the focus operation, I think it is important to check the whole operation and not just the name of the operation.

Reply:
In my previous PR to torch-mlir, I was asked to remove the unit test (https://github.com/llvm/torch-mlir/pull/1795#discussion_r1068121900)
They have a policy to only add unit tests when the code is complicated enough (not just replacing one op by a sequence of other ops).
But then torch-mlir requires to have the change covered by an end-to-end test. (I.e. what Liam said about extending TOSA_PASS_SET).

Reply:
I've tried to add the following IR to check the exact dimensions for the padding sizes. But FileCheck keeps reporting error "test/Conversion/TorchToTosa/constant_pad_nd.mlir:5:64: error: missing closing "]" for regex variable"

%2 = "tosa.const"() {value = dense<[[0, 0], [0, 0], [2, 3], [0, 1]]> : tensor<4x2xi64>} : () -> tensor<4x2xi64>

it seems that it cannot distinguish ] from the regEx. Tried to use escape symbol "\", but didn't help either.

Reply:
@lisaliu1 - going back to the point @mgehre-amd riased, I see this file is a single addition beside `test/Conversion/TorchToTosa/basic.mlir`, which suggests it is out of keeping with how they add tests. In https://github.com/llvm/torch-mlir/pull/1790 both the addition to `TorchToTosa/basic.mlir` and to `TOSA_PASS_SET` was accepted, but I expect only extending `TOSA_PASS_SET` is enough as it effectively adds the same cases, so I recommend against adding these `FileCheck` tests unless an upstream reviewer requests them..